### PR TITLE
Decomp bag, bag_view, itemtool, heap, pokewalker

### LINKED ISSUE #30: knarc segfaults if there are no input files
To reproduce:
- Create an empty directory (except maybe for .*ignore files and any file pattern matching the .*ignore spec)
- Run knarc in pack mode on that directory

Expectation:
knarc exits or warns with a useful message

Observation:
knarc throws SIGSEGV (11)

### PR DESCRIPTION
**Pull request by PikalaxALT**
Resolves #30

### CODE REVIEW

File: src/pm_version.c [hunk 1]
```diff
@@ -1,4 +1,3 @@
-#include "global.h"
```
Comment (tgsm) [inline src/pm_version.c:1]: Extra newline

File: src/item.c [hunk 2]
```diff
@@ -0,0 +1,1046 @@
...
+    return &itemData[itemDataIdx];
+}
+
+BOOL ItemIdIsNotJohtoBall(u16 itemId) {
+    // stupid hack to get it to match
+    if ((u16)(itemId + (u16)-ITEM_FAST_BALL) > (ITEM_PARK_BALL - ITEM_FAST_BALL)) {
+        return TRUE;
+    } else {
+        return FALSE;
+    }
```
Comment (tgsm) [inline src/item.c:1045]:
```suggestion
    return ((u16)(itemId + (u16)-ITEM_FAST_BALL) > (ITEM_PARK_BALL - ITEM_FAST_BALL));
```

File: src/seal_case.c [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+    num = SealCase_CountSealOccurrenceInUse(sealCase, sealId - SEAL_MIN);
+    total = num + sealCase->inventory.seals[sealId - SEAL_MIN];
+
+    if (quantity < 0) {
+        if (sealCase->inventory.seals[sealId - SEAL_MIN] + quantity >= 0) {
+            return TRUE;
+        } else {
+            return FALSE;
+        }
```
Comment (tgsm) [inline src/seal_case.c:146]:
```suggestion
        return (sealCase->inventory.seals[sealId - SEAL_MIN] + quantity) >= 0;
```

File: src/seal_case.c [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    } else {
+        if (total + quantity <= SEAL_INVENTORY_QMAX) {
+            return TRUE;
+        } else {
+            return FALSE;
+        }
```
Comment (tgsm) [inline src/seal_case.c:152]:
```suggestion
        return (total + quantity) <= SEAL_INVENTORY_QMAX;
```

File: src/seal_case.c [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    }
+
+    return TRUE;
```
Comment (tgsm) [inline src/seal_case.c:155]: this line gets optimized out, so we don't need it.

File: include/save.h [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+#define SAVE_UNK_8                8
+#define SAVE_UNK_9                9
+#define SAVE_UNK_10              10
+#define SAVE_UNK_11              11
+#define SAVE_DRESSUP_DATA        12
+#define SAVE_MAILBOX             13
+#define SAVE_UNK_14              14
+#define SAVE_UNK_15              15
+#define SAVE_GAMESTATS           16
+#define SAVE_SEAL_CASE              17
```
Comment (tgsm) [inline include/save.h:23]: spacing

File: include/mail.h [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+};
+
+typedef struct Mail
+{
+    u32 author_otId; // author otid
+    u8 author_gender; // author gender
+    u8 author_language; // language
+    u8 author_version; // version
+    u8 mail_type; // mail type
+    u16 author_name[OT_NAME_LENGTH + 1]; // author name
```
Comment (tgsm) [inline include/mail.h:22]: these comments are kinda redundant

**Review by tgsm — CHANGES_REQUESTED** [review]

File: src/seal_case.c (hunk 3 above)
Comment (PikalaxALT) [reply src/seal_case.c:146]: Rejecting change to maintain similarity with the above two functions.

File: src/item.c (hunk 2 above)
Comment (PikalaxALT) [reply src/item.c:1045]: Rejecting change to maintain consistency with other files

File: src/seal_case.c (hunk 4 above)
Comment (PikalaxALT) [reply src/seal_case.c:152]: Rejecting change to maintain similarity with the above two functions.

**Review by tgsm — APPROVED** [review]

**Pull request merged**